src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double GROWING_COEFFICIENT = 1.5;
+    private static final int DEFAULT_LENGTH = 10;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = growIfArrayFull();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size()) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index);
+        }
+        if (size == elementData.length) {
+            elementData = growIfArrayFull();
+        }
+        if (index != size) {
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        }
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() > elementData.length) {
+            growIfArrayFull();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIfArrayListOutOfBounds(index);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIfArrayListOutOfBounds(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIfArrayListOutOfBounds(index);
+        T element = elementData[index];
+        System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
+        size--;
+        return element;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size(); i++) {
+            if (elementData[i] == null && element == null
+                    || elementData[i] != null && elementData[i].equals(element)) {
+                remove(i);
+                return element;
+            }
+        }
+        throw new NoSuchElementException("No such element in list: " + element);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        if (size() > 0) {
+            return false;
+        }
+        return true;
```

Comment:
Consider simplifying the code. You can directly return the result of the condition instead of using if/else or ternary operation to return true/false. For example:

// Instead of this
if (condition) {
    return true;
} else {
    return false;
}
// You can do this
return condition;

This makes the code cleaner and more straightforward.